lib/grinder_tools.dart

```diff
@@ -139,10 +139,10 @@ Future runProcessAsync(GrinderContext context, String executable,
  * A default implementation of an `init` task. This task verifies that the grind
  * script is executed from the project root.
  */
-void defaultInit(GrinderContext context) {
+void defaultInit([GrinderContext context]) {
```

Comment:
I take that back, why make users go through two separate migrations.  I think we should move the parameters that can't be made optional now.  Then they can update all their code when moving to 0.7.0, but at the same time for the newly optional params, they are not required to update yet if they don't want to, but will get a nice deprecation message if we mark as such.  I'd also be fine with just removing the newly optional ones too.